internal/tofu/evaluate.go

```diff
@@ -943,6 +943,37 @@ func (d *evaluationStateData) GetTerraformAttr(addr addrs.TerraformAttr, rng tfd
 	}
 }
 
+func (d *evaluationStateData) GetTofuAttr(addr addrs.TofuAttr, rng tfdiags.SourceRange) (cty.Value, tfdiags.Diagnostics) {
```

Comment:
I think it may be best to have a common extracted function between this and GetTerraofmrAttr so that if we make any changes to one, the other does not regress.

Reply:
OK. 

extract a common method `getTerraformOrTofuAttr`

In order to ensure the accuracy of the information in diagnostic, I defined an additional ObjectName method on TerraformAttr and TofuAttr.